nixos/modules/services/web-apps/plausible.nix

```diff
@@ -0,0 +1,270 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.plausible;
+
+  envSecrets = ''
+    export ADMIN_USER_PWD="$(<${cfg.adminUser.passwordFile})"
+    export SECRET_KEY_BASE="$(<${cfg.server.secretKeybaseFile})"
+    export RELEASE_TMP=/var/lib/plausible/tmp
+    ${optionalString (cfg.mail.smtp.passwordFile != null) ''
+      export SMTP_USER_PWD="$(<${cfg.mail.smtp.passwordFile})"
+    ''}
+  '';
+in {
+  options.services.plausible = {
+    enable = mkEnableOption "plausible";
+
+    adminUser = {
+      name = mkOption {
+        default = "admin";
+        type = types.str;
+        description = ''
+          Name of the admin user that plausible will created on initial startup.
+        '';
+      };
+
+      email = mkOption {
+        type = types.str;
+        example = "admin@localhost";
+        description = ''
+          Email-address of the admin-user.
+        '';
+      };
+
+      passwordFile = mkOption {
+        type = types.either types.str types.path;
+        description = ''
+          Path to the file which contains the password of the admin user.
+        '';
+      };
+
+      activate = mkEnableOption "activating the freshly created admin-user";
+    };
+
+    database = {
+      clickhouse = {
+        setup = mkEnableOption "creating a clickhouse instance" // { default = true; };
+        url = mkOption {
+          default = "http://localhost:8123/default";
+          type = types.str;
+          description = ''
+            The URL to be used to connect to <package>postgres</package>. The format
+            is described in <link xlink:href="https://hexdocs.pm/ecto/Ecto.Repo.html#module-urls">
+            the elixir docs</link>.
+          '';
+        };
+      };
+      postgres = {
+        setup = mkEnableOption "creating a postgresql instance" // { default = true; };
+        dbname = mkOption {
+          default = "plausible";
+          type = types.str;
+          description = ''
+            Name of the database to use.
+          '';
+        };
+        socket = mkOption {
+          default = "/run/postgresql";
+          type = types.str;
+          description = ''
+            Path to the UNIX domain-socket to communicate with <package>postgres</package>.
+          '';
+        };
+      };
+    };
+
+    server = {
+      disableRegistration = mkOption {
```

Comment:
Is this the upstream option naming? I generally prefer positive names (e.g. `allowRegistration` or `enableRegistration`) because less negation means less cognitive overhead.

Reply:
I'm all for positivity as well and also disagree a bit with their naming (including the remark you've had) :)).

However I'm generally a fan of maintaining consistency with upstream if possible and given that we also have things like RFC42 now, this seems to be a general trend (note that I've decided against that here because I haven't found a nice way to transform the options into environment vars while properly securing secrets and there aren't too many options so I guess that's fine).